routing/routing.go

```diff
@@ -182,15 +182,7 @@ type LBAlgorithm interface {
 type LBContext struct {
 	Request *http.Request
 	Route   *Route
-}
-
-// NewLBContext is used to create a new LBContext, to pass data to the
-// load balancer algorithms.
-func NewLBContext(r *http.Request, rt *Route) *LBContext {
```

Comment:
this is a breaking API change.
We should try to omit these if we can.
Maybe better to let it and mar as deprecated and point to the new usage.
We can get rid of it in v1

Reply:
Yes, but only if anyone uses this API. The fix is straightforward - create `LBContext` instead of calling this constructor. Extending `LBContext` in backward compatible way is possible while extending this constructor is not so hopefully this is a one-time change for the the users (if any) to accommodate to. We may also think about considering all APIs as experimental by default until they are stable enough to require no extension and marked explicitly as stable, otherwise we will continue to face such extension impediments.

> We can get rid of it in v1

IMO there is no timeline for v1 and there is a risk to end up with a fork like Python 2/3 so v1 may very-well never happen.


I can keep the constructor and create `LBContext` directly if we really want to stick to zero user (if any) disturbance policy, just thought we may discuss this.

Reply:
Let's ask @aryszka as a 3rd person, I am fine with following his judgement. :)

Reply:
The only problem is that we don't know and should not make assumptions about by whom and how it is used. Pragmatism is of course welcome, and sometimes even necessary, but in this particular case, it is very easy to do the right thing, which in my opinion is to keep the initializer and mark it deprecated.

For us it's also a learning: in Go we don't need to create useless initializers where all the fields of a structure are exported and the initializer has no valuable logic.